internal/error.go

```diff
@@ -241,9 +250,9 @@ var (
 )
 
 // NewApplicationError create new instance of *ApplicationError with message, type, and optional details.
-func NewApplicationError(message, errType string, nonRetryable bool, cause error, details ...interface{}) *ApplicationError {
+func NewApplicationError(msg, errType string, nonRetryable bool, cause error, details ...interface{}) *ApplicationError {
```

Comment:
can we not reuse the var type? 
like `msg, errType string` vs `msg string, errType string`
former is hard to parse?